### PR TITLE
Feature/house product

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/service/aggregates/ExtraBuilding.kt [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
-import javax.persistence.EnumType
-import javax.persistence.Enumerated
+import com.hedvig.productPricing.service.web.dto.ExtraBuildingDTO
+import java.util.*
+import javax.persistence.*
 
-@Embeddable
-data class ExtraBuilding(
+@Entity
+class ExtraBuilding(
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/productPricing/service/aggregates/ExtraBuilding.kt]: data class?

File: src/main/java/com/hedvig/productPricing/service/aggregates/MemberAggregate.java [hunk 2]
```diff
@@ -133,24 +152,12 @@ public UUID createModifiedProduct(CreateModifiedProductCommand command) {
       command.getAncillaryArea(),
       command.getYearOfConstruction(),
       command.getNumberOfBathrooms(),
-      command.getExtraBuildings(),
+      command.getExtraBuildings() != null
+        ? command.getExtraBuildings().stream().map(ExtraBuilding.Companion::from).collect(Collectors.toList())
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/productPricing/service/aggregates/MemberAggregate.java]: Duplicated code extract to method?

File: src/main/java/com/hedvig/productPricing/service/web/dto/InsuranceStatusDTO.java [hunk 3]
```diff
@@ -133,7 +132,9 @@ public InsuranceStatusDTO(ProductEntity productEntity, String certificateUrl, Li
     this.ancillaryArea = productEntity.ancillaryArea;
     this.yearOfConstruction = productEntity.yearOfConstruction;
     this.numberOfBathrooms = productEntity.numberOfBathrooms;
-    this.extraBuildings = productEntity.extraBuildings;
+    this.extraBuildings = productEntity.extraBuildings != null
+                            ? productEntity.extraBuildings.stream().map(ExtraBuildingDTO.Companion::from).collect(Collectors.toList())
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/productPricing/service/web/dto/InsuranceStatusDTO.java]: Same here duplicated in kotlin we could have a extension for this

File: src/main/java/com/hedvig/productPricing/service/web/dto/ExtraBuildingDTO.kt [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        ?: getDefaultDisplayName(type)
+      ExtraBuildingType.SAUNA -> localizationService.getText(locale, "EXTRA_BUILDING_DISPLAY_NAME_SAUNA")
+        ?: getDefaultDisplayName(type)
+      ExtraBuildingType.BARN -> localizationService.getText(locale, "EXTRA_BUILDING_DISPLAY_NAME_BARN")
+        ?: getDefaultDisplayName(type)
+      ExtraBuildingType.BOATHOUSE -> localizationService.getText(locale, "EXTRA_BUILDING_DISPLAY_NAME_BOATHOUSE")
+        ?: getDefaultDisplayName(type)
+      ExtraBuildingType.OTHER -> localizationService.getText(locale, "EXTRA_BUILDING_DISPLAY_NAME_OTHER")
+        ?: getDefaultDisplayName(type)
+    }
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/productPricing/service/web/dto/ExtraBuildingDTO.kt]: Could probably replace this with ```
private fun getDisplayName(type: ExtraBuildingType, localizationService: LocalizationService, locale: Locale) = 
localizationService.getText(locale, "EXTRA_BUILDING_DISPLAY_NAME_${type.name}") ?: getDefaultDisplayName(type)
```

Comment (vonElfvin) [reply src/main/java/com/hedvig/productPricing/service/web/dto/ExtraBuildingDTO.kt]: Done, great improvement

**Pull request merged**